godel_surface_detection/include/godel_surface_detection/data_coordinator/data_coordinator.h

```diff
@@ -0,0 +1,391 @@
+#ifndef DATA_COORDINATOR_H
+#define DATA_COORDINATOR_H
+
+#include <pcl/PolygonMesh.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <geometry_msgs/PoseArray.h>
+#include <ros/io.h>
+
+namespace godel_surface_detection
+{
+namespace data
+{
+  // Pose Types
+  const static int BLEND_POSE_TYPE = 1;
+  const static int SCAN_POSE_TYPE = 2;
+
+  // Cloud Types
+  const static int INPUT_CLOUD_TYPE = 1;
+  const static int SURFACE_CLOUD_TYPE = 2;
+  const static int BOUNDARY_CLOUD_TYPE = 3;  // for future use
+
+  // Common Error Strings
+  const static std::string UNABLE_TO_FIND_RECORD_ERROR = "Unable to get record with specified id";
+
+  /**
+   * @brief A structure containing features pertinent to surface detection.
+   */
+  struct SurfaceDetectionRecord
+  {
+    public:
+      int id_;
+      std::string surface_name_;
+      pcl::PointCloud<pcl::PointXYZRGB> input_cloud_;
+      pcl::PolygonMesh surface_mesh_;
+      pcl::PointCloud<pcl::PointXYZRGB> surface_cloud_;
+      std::vector<std::pair<std::string, geometry_msgs::PoseArray>> edge_pairs_;
+      geometry_msgs::PoseArray blend_poses_;
+      geometry_msgs::PoseArray scan_poses_;
+  };
+
+
+  /**
+   * @brief Class to handle acquiring and diseminating data relevant to surface detection features
+   */
+  class DataCoordinator
+  {
+  private:
+    int id_counter_ = 0;
+    std::vector<SurfaceDetectionRecord> records_;
+
+    /**
+     * @brief getNextID return a unique ID
+     * @return
+     */
+    int getNextID()
+    {
+      return ++id_counter_;
+    }
+
+
+    /**
+     * @brief printIds
+     * @return a formatted string containing all current record IDs
+     */
+    std::string printIds()
+    {
+      std::stringstream ss;
+      std::string separator = "[";
+      for(const auto& rec : records_)
+      {
+        ss << separator << rec.id_;
+        separator = ", ";
+      }
+      ss << "]";
+      return ss.str();
+    }
+
+
+  public:
+    //! Default Constructor
+    DataCoordinator()
+    {
+      id_counter_ = 0;
+      if(records_.size() > 0)
+        records_.clear();
+    }
+
+    /**
+     * @brief init Initializes DataCoordinator object
+     * @return success of operation
+     */
+    bool init()
```

Comment:
It was intended more as a way to clean the object (not sure I had a good purpose there) and to mimic the style of InteractiveSurfaceServer, SurfaceBlendingService, and RobotScan. 

The other 3 classes call init() in surface_blending_service.cpp

`if (surface_detection_.init() && robot_scan_.init() && surface_server_.init() && data_coordinator_.init())
  {...`

And having the init() signals to the reader that the object is ready for use. It is, however, logically superfluous, as I don't use it elsewhere, and I can remove it with few changes.